### PR TITLE
prevent multiple calls to disconnect for serial and tcp

### PR DESCRIPTION
**Pull request by Apollon77**
as discussed in one of the node-mbus issues here is my change to prevent multiple "disconnect" calls to make problems by checking it.

### CODE REVIEW

**Comment by lategoodbye** [comment]
Does this prevent the segmentation faults?

**Comment by Apollon77** [comment]
We tracked the topic down to a topic in my own code where it happened that the mbus_disconnect method was called twice and then the segfault happened at thesecond call of "close(handle->fd);" So yes this change makes sure that "close" is only called once for a filedescriptor at all.

It simply make sure "bad user code" do not break anything

File: mbus/mbus-serial.c [hunk 1]
```diff
@@ -191,7 +191,13 @@ mbus_serial_disconnect(mbus_handle *handle)
         return -1;
     }
 
+    if (handle->fd == -1)
+    {
```
Comment (lategoodbye) [inline mbus/mbus-serial.c]: From my understand a valid fd is >= 0, so could you please change both if-statements to:

`if (handle->fd < 0)`

Comment (Apollon77) [reply mbus/mbus-serial.c]: I was not able to find anything instead of "set a invalid number like -1" :-)

Yes I iwll change that

Comment (lategoodbye) [reply mbus/mbus-serial.c]: This is not what i wrote < 0 and mean (both = serial and tcp).

Comment (Apollon77) [reply mbus/mbus-serial.c]: Sorry overread the "Both"

Comment (lategoodbye) [reply mbus/mbus-serial.c]: And why did you ignored my suggestion in the first comment?

Comment (Apollon77) [reply mbus/mbus-serial.c]: SImply because I was btoo blind :-(

Comment (Apollon77) [reply mbus/mbus-serial.c]: done

Comment (lategoodbye) [reply mbus/mbus-serial.c]: Good. Could you please check this changes against your CI before i merge this?

Comment (Apollon77) [reply mbus/mbus-serial.c]: https://travis-ci.org/Apollon77/libmbus/builds/359475199

Linux already done, macos currently has big backlog, so can take a while but should be ok too.

BTW: Travis-CI file is already in your directory, to enable that for you too are kind of 3 klicks :-) Then it runs automatically with each commit and PR. Simply signup on travis-ci.org with your Github account

Comment (lategoodbye) [reply mbus/mbus-serial.c]: Linux is sufficient. Unfortunately i'm not owner of this repository (just contributor), so i can't play with the settings here.

Comment (Apollon77) [reply mbus/mbus-serial.c]: Aahhh ... if you can contact them ... it's worth it :)

**Pull request merged**